build_tools/android/common/benchmark_description.py

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities for describing Android benchmarks.
+
+This file provides common and structured representation of Android devices,
+benchmark definitions, and benchmark result collections, so that they can be
+shared between different stages of the same benchmark pipeline.
+"""
+
+import json
+import re
+import subprocess
+
+from dataclasses import dataclass
+from typing import Any, Dict, Sequence
+
+__all__ = ["AndroidDeviceInfo", "BenchmarkInfo", "BenchmarkResults"]
+
+# A map for IREE driver names. This allows us to normalize driver names like
+# mapping to more friendly ones and detach to keep driver names used in
+# benchmark presentation stable.
+IREE_DRIVER_NAME_MAP = {
+    "iree_llvmaot": "IREE-Dylib",
+    "iree_vmvx": "IREE-VMVX",
+    "iree_vulkan": "IREE-Vulkan",
+}
+
+
+def execute(args: Sequence[str],
+            capture_output: bool = False,
+            treat_io_as_text: bool = True,
+            verbose: bool = False,
+            **kwargs) -> subprocess.CompletedProcess:
+  """Executes a command."""
+  if verbose:
+    cmd = " ".join(args)
+    print(f"cmd: {cmd}")
+  return subprocess.run(args,
+                        check=True,
+                        capture_output=capture_output,
+                        text=treat_io_as_text,
```

Comment:
I think I commented on this elsewhere, but this kwarg only got added in 3.7. Please use the awfully-named `universal_newlines` for backward compatibility

```suggestion
                        universal_newlines=treat_io_as_text,
```